roles/velero/templates/registry_gcp.yml.j2

```diff
@@ -0,0 +1,72 @@
+---
+{% for bsl in backup_storage_locations %}
+{% if bsl.provider == 'gcp' %}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: {{ velero_namespace }}
+  name: gcp-registry
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      component: gcp-registry
+  template:
+    metadata:
+      labels:
+        component: gcp-registry
+    spec:
+      restartPolicy: Always
+      containers:
+      - image: quay.io/konveyor/registry:latest
+        name: gcp-registry-container
+        ports:
+        - containerPort: 5000
+          protocol: TCP
+        env:
+          - name: REGISTRY_STORAGE
```

Comment:
Same comment as below. Let's use one template and then only set the GCP env vars if we determine its a GCP registry.

Reply:
Kept deployment templates different for each provider as discussed and moved out the svc + route creation.